src/test/java/core/basesyntax/service/impl/TransactionValidatorImplTest.java

```diff
@@ -0,0 +1,135 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.exception.InvalidDataException;
+import core.basesyntax.service.interfaces.TransactionValidator;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class TransactionValidatorImplTest {
+    private static final String HEADING = "type,fruit,quantity";
+    private static final String INVALID_HEADING = "thur,fruitstor,bokuyt";
+    private static final String EMPTY_HEADING = "";
+    private static final String VALID_RECORD = "b,banana,100";
+    private static final String VALID_PURCHASE_RECORD = "p,banana,100";
+    private static final String VALID_SUPPLY_RECORD = "s,banana,100";
+    private static final String VALID_RETURN_RECORD = "r,banana,100";
+    private static final String INVALID_RECORD_FORMAT = "banana,b,100";
+    private static final String INVALID_RECORD_OPERATION = "k,banana,100";
+    private static final String INVALID_RECORD_QUANTITY = "b,banana,-50";
+
+    private TransactionValidator validator;
+
+    @BeforeEach
+    void setUp() {
+        validator = new TransactionValidatorImpl();
+    }
+
+    @Test
+    void validate_OK() {
+        String input = HEADING + System.lineSeparator()
+                + VALID_RECORD + System.lineSeparator();
+        validator.validate(input);
+        Assertions.assertDoesNotThrow(() -> new InvalidDataException("Data is invalid"));
```

Comment:
Are you sure that you are using this assertion correctly? You should pass runnable with executable, not exception, similar to what you did in `assertThrows`

Reply:
Not resolved